x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/rules_table/index.tsx

```diff
@@ -86,12 +90,19 @@ export const MigrationRulesTable: React.FC<MigrationRulesTableProps> = React.mem
     const [sortField, setSortField] = useState<keyof RuleMigrationRule>(DEFAULT_SORT_FIELD);
     const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>(DEFAULT_SORT_DIRECTION);
     const [searchTerm, setSearchTerm] = useState<string | undefined>();
+    const [isMissingIndexPatternFlyoutOpen, setIsMissingIndexPatternFlyoutOpen] = useState(false);
+    const [missingIndexPatternSelected, setMissingIndexPatternSelected] = useState(0);
+    const [installTranslatedRulesSelected, setInstallTranslatedRulesSelected] = useState(0);
+    const [reprocessFailedRulesSelected, setReprocessFailedRulesSelected] = useState(0);
```

Comment:
These guys are specific to the bulk actions. It looks like they are defined out of place. Consider moving them inside the `BulkActions`.
Actually, I think the `BulkActions` component should be responsible for implementing all the logic related to the actions, including flyouts, modals and so on. So we separate concerns and leave this component (already pretty big) free of actions-related logic.

Probably, these props should be enough.

```ts
                    <BulkActions
                      isTableLoading={isRulesLoading}
                      migrationStats={migrationStats}
                      translationStats={translationStats}
                      selectedRules={selectedMigrationRules}
                    />
```

Reply:
The actions may be better defined here since they play with the selected and loaders, but the rest can probably be moved inside.